packages/main/src/plugin/container-registry.ts

```diff
@@ -2334,12 +2337,35 @@ export class ContainerProviderRegistry {
     } else {
       const dockerInfo = await provider.api.info();
       return {
+        engineId: provider.id,
+        engineName: provider.name,
+        engineType: provider.connection.type,
         cpus: dockerInfo.NCPU,
         memory: dockerInfo.MemTotal,
       };
     }
   }
 
+  async listInfos(options?: containerDesktopAPI.ListInfosOptions): Promise<containerDesktopAPI.ContainerEngineInfo[]> {
+    let providers: InternalContainerProvider[];
+    if (options?.provider === undefined) {
+      providers = Array.from(this.internalProviders.values());
+    } else {
+      providers = [this.getMatchingContainerProvider(options?.provider)];
+    }
+    const infos = await Promise.all(
+      Array.from(providers).map(async provider => {
+        try {
+          return await this.info(provider.id);
+        } catch (error) {
+          console.log('error getting info for engine', provider.name, error);
+          return [];
+        }
+      }),
+    );
+    return infos.flat();
```

Comment:
I'm not sure here about the usage of array and then flat
vs filtering out undefined
return Array.from(providers).map(provider => return info(provider).filter(p => !p)

Reply:
I cannot get the type checker detect that `filter(info => info)` does not return undefined values. The result is always of type `(containerDesktopAPI.ContainerEngineInfo | undefined)[]`

```
    const infos = await Promise.all(
      Array.from(providers).map(async provider => {
        try {
          return await this.info(provider.id);
        } catch (error) {
          console.log('error getting info for engine', provider.name, error);
          return undefined;
        }
      }).filter(async info => info),
    );
    return infos;
```

Or do I misunderstand what you propose?

Reply:
the trick is to use the cast after the filter call as you know it's filtering out undefined

=> `.filter(async info => info) as ContainerEngineInfo`

Reply:
`filter((item): info is ContainerEngineInfo => !!info)`